static/app/components/assigneeSelectorDropdown.tsx

```diff
@@ -0,0 +1,543 @@
+import {Fragment} from 'react';
+import styled from '@emotion/styled';
+import uniqBy from 'lodash/uniqBy';
+
+import {openInviteMembersModal} from 'sentry/actionCreators/modal';
+import ActorAvatar from 'sentry/components/avatar/actorAvatar';
+import SuggestedAvatarStack from 'sentry/components/avatar/suggestedAvatarStack';
+import {Button} from 'sentry/components/button';
+import {Chevron} from 'sentry/components/chevron';
+import {
+  CompactSelect,
+  type SelectOption,
+  type SelectOptionOrSection,
+} from 'sentry/components/compactSelect';
+import IdBadge from 'sentry/components/idBadge';
+import ExternalLink from 'sentry/components/links/externalLink';
+import LoadingIndicator from 'sentry/components/loadingIndicator';
+import {Tooltip} from 'sentry/components/tooltip';
+import {IconAdd, IconUser} from 'sentry/icons';
+import {t, tct, tn} from 'sentry/locale';
+import ConfigStore from 'sentry/stores/configStore';
+import MemberListStore from 'sentry/stores/memberListStore';
+import ProjectsStore from 'sentry/stores/projectsStore';
+import {useLegacyStore} from 'sentry/stores/useLegacyStore';
+import {space} from 'sentry/styles/space';
+import type {Actor, Group, SuggestedOwnerReason, Team, User} from 'sentry/types';
+import {buildTeamId} from 'sentry/utils';
+
+const suggestedReasonTable: Record<SuggestedOwnerReason, string> = {
+  suspectCommit: t('Suspect Commit'),
+  ownershipRule: t('Ownership Rule'),
+  projectOwnership: t('Ownership Rule'),
+  // TODO: codeowners may no longer exist
+  codeowners: t('Codeowners'),
+};
+
+export type AssignableEntity = {
+  assignee: User | Actor;
+  id: string;
+  type: Actor['type'];
+  suggestedAssignee?: SuggestedAssignee;
+};
+
+export type SuggestedAssignee = Actor & {
+  assignee: AssignableTeam | User;
+  suggestedReason: SuggestedOwnerReason;
+  suggestedReasonText?: React.ReactNode;
+};
+
+type AssignableTeam = {
+  display: string;
+  email: string;
+  id: string;
+  team: Team;
+};
+
+export interface AssigneeSelectorDropdownProps {
+  group: Group;
```

Comment:
This will need to be generalized for alerts and user feedback in the future so we can replace the assignee dropdown on their respective pages